modules/core/src/v2/coins/eos.ts

```diff
@@ -822,23 +886,32 @@ export class Eos extends BaseCoin {
         memo: destinationAddressDetails.memoId,
       });
 
-      const transaction = yield eosClient.transaction({ actions: [transferAction] }, { sign: false, broadcast: false });
+      let serializedTransaction;
+      const tx = { actions: [transferAction] };
+      try {
+        serializedTransaction = yield api.transact({ ...tx, ...headers }, { sign: false, broadcast: false });
+      } catch (e) {
+        throw Error('Eos API error: Could not build transaction');
+      }
+
+      // generate transactionId
+      const transactionId = Eos.createTransactionIdHex((serializedTransaction as any).serializedTransaction);
 
-      const serializedTransaction = self.serializeTransaction(eosClient, transaction);
+      const serializedTransactionHex = Buffer.from(serializedTransaction.serializedTransaction).toString('hex');
       const txObject = {
         transaction: {
           compression: 'none',
-          packed_trx: serializedTransaction,
+          packed_trx: serializedTransactionHex,
           signatures: [] as string[],
         },
-        txid: (transaction as any).transaction_id,
+        txid: transactionId,
         recoveryAmount: accountBalance,
         coin: self.getChain(),
         txHex: '',
       };
       const signableTx = Buffer.concat([
         Buffer.from(self.getChainId(), 'hex'), // The ChainID representing the chain that we are on
-        Buffer.from(serializedTransaction, 'hex'), // The serialized unsigned tx
+        Buffer.from(serializedTransactionHex, 'hex'), // The serialized unsigned tx
```

Comment:
seems a bit strange to convert to a buffer, then to a hex string, then use the hex string, then later convert that hex string back to a buffer 🤔